src/main/java/core/basesyntax/strategy/CalculateStrategy.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Map;
+
+public class CalculateStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> correspondenceTable;
+
+    public CalculateStrategy(Map<FruitTransaction.Operation,
+            OperationHandler> correspondenceTable) {
+        this.correspondenceTable = correspondenceTable;
+    }
+
+    public void calculate(FruitTransaction fruitTransaction,
+                          FruitTransaction.Operation operation) {
+        OperationHandler handler = correspondenceTable.get(operation);
+        handler.handle(fruitTransaction);
+    }
```

Comment:
Strategy aim is to provide proper instrument for solving task, not to solve it by itself
```suggestion
    public OperationHandler getHandler(FruitTransaction.Operation operation) {
        return correspondenceTable.get(operation);
    }
```